electron/src/routes.ts

```diff
@@ -458,6 +461,29 @@ module.exports = {
         }).catch(callback);
     },
 
+    checkForPlatformUpdates: (data: {}, callback) => {
+
+        const api = new GitHubClient();
+
+        return api.releases().get().then((result: Array<any>) => {
+
+            let hasUpdate = null;
+
+            if (result && result.length) {
+                result.sort((a, b) => semver.gt(a, b));
+
+                const latestRelease = result[0];
+
+                if (semver.gt(latestRelease.tag_name, app.getVersion())) {
+                    hasUpdate = latestRelease;
+                }
+            }
+
+            callback(null, hasUpdate);
+
+        }, callback).catch(callback);
```

Comment:
Duplicate catching. Second argument to `.then()` is an error handler.
`.then(result => callback(null, result), callback)` is same as `.then(result => callback(null, result)).catch(callback)`